AC_TrackCycle/TrackCyclerForm.cs

```diff
@@ -491,22 +492,29 @@ private void button_addToWhiteList_Click(object sender, EventArgs e)
         {
             if(textBox_driver_guid.Text != null && textBox_driver_guid.Text != "")
             {
-                whiteList.Add(textBox_driver_guid.Text);
-                listBox_whitelist.DataSource = whiteList.ToList();
-                listBox_whitelist.Refresh();
-                if (File.Exists(AppDomain.CurrentDomain.BaseDirectory + "\\whitelist.txt"))
+                if (whiteList.Contains(textBox_driver_guid.Text))
                 {
-                    File.Delete(AppDomain.CurrentDomain.BaseDirectory + "\\whitelist.txt");
+                    MessageBox.Show("Already in list");
                 }
-                FileStream f = File.Create(AppDomain.CurrentDomain.BaseDirectory + "\\whitelist.txt");
-                var offset = 0;
-                foreach (var guid in whiteList)
+                else
                 {
-                    byte[] toWrite = Encoding.ASCII.GetBytes(guid + Environment.NewLine);
-                    f.Write(toWrite, 0, toWrite.Length);
-                    offset += toWrite.Length;
+                    whiteList.Add(textBox_driver_guid.Text);
+                    listBox_whitelist.DataSource = whiteList.ToList();
+                    listBox_whitelist.Refresh();
+                    if (File.Exists(whiteListStoragePath))
+                    {
+                        File.Delete(whiteListStoragePath);
+                    }
+                    FileStream f = File.Create(whiteListStoragePath);

```

Comment:
could write file with StreamWriter, makes it a little easier. But since it's the same result, I'll leave it for now.

            using (var streamWriter = new StreamWriter(whiteListStoragePath, Encoding.ASCII))
            {
                foreach (var guid in whiteList)
                {
                    streamWriter.WriteLine(guid);
                }
            }